Lib/http/server.py

```diff
@@ -331,6 +332,7 @@ def parse_request(self):
                     "Bad HTTP/0.9 request type (%r)" % command)
                 return False
         self.command, self.path = command, path
+        self.path = re.sub(r'(^(/|\\)+)', '/', self.path)
```

Comment:
I would prefer to restrict the security fix to / character. For bpo-43223, only / matters.

If there is an issue with \ on Windows, I would prefer to treat it differently.

You added the following comment on GitHub:

"bpo-43223: The purpose of replacing // with / is to provide protection against open redirect attack reside within http.server module which can be triggered due to the presence of // at the beginning of the path."

But I suggested to add it in the code, so future readers don't have to dig into the 24000+ pull requests to understand why http.server replaces // with / :-)

About the comment, I think that the important information is that `/path` is treated as a **relative** URL, whereas `//path` is treated as an **absolute** URL (like `http://path` with no scheme).

Reply:
Okay, I'm on it :)

Reply:
> About the comment, I think that the important information is that `/path` is treated as a **relative** URL, whereas `//path` is treated as an **absolute** URL (like `http://path` with no scheme).

Yes, you're absolutely right. Just because of the `//` at the beginning of the path the web client presumes it to be an absolute path with no scheme and redirect it.

Reply:
Updated the code and pushed the commit. Please review and mention if some other changes are required

Reply:
Just want to mention, I tweaked regex a little bit more:
```python
self.path = re.sub(r'(/)+', '/', self.path)
```

Now rather than replacing `//` from the beginning it'll be replacing from the entire path if it matches any. Like:
`'//attacker.com//...' -> '/attacker.com/..'`